pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -385,9 +385,9 @@ func (jm *ControllerV2) updateCronJob(old interface{}, curr interface{}) {
 	// if the change in schedule results in next requeue having to be sooner than it already was,
 	// it will be handled here by the queue. If the next requeue is further than previous schedule,
 	// the sync loop will essentially be a no-op for the already queued key with old schedule.
-	if oldCJ.Spec.Schedule != newCJ.Spec.Schedule {
+	if oldCJ.Spec.Schedule != newCJ.Spec.Schedule || oldCJ.Spec.TimeZone != newCJ.Spec.TimeZone {
```

Comment:
Maybe we will need helper function for comparing time zone values?
"" is equivalent to "UTC", for example.

Reply:
Empty isn't equal to UTC, the former defaults to whatever kube-controller-manager process is running with, the latter explicitly sets UTC, no need to change here anything.

Reply:
One missing thing is using feature gate, this field is alpha based on the [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3140-TimeZone-support-in-CronJob) so you need explicitly check if the feature is on or off and react accordingly.